src/net/http/UploadFile.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef NET_HTTP_UPLOADFILE_HPP
+#define NET_HTTP_UPLOADFILE_HPP
+
+#include "Job/Job.hpp"
+#include "system/Path.hpp"
+
+#include <array>
+#include <cstddef> // for std::byte
+#include <tchar.h>
+
+class OperationEnvironment;
+class CurlGlobal;
+
+namespace Net {
+
+/**
+ * Upload a URL into the specified file.
+ *
+ * Throws on error.
+ *
+ */
+void UploadFile(CurlGlobal &curl, const TCHAR *url,
+                    const TCHAR *username, const TCHAR *password,
+                    Path path, char* buffer, size_t buffer_size, OperationEnvironment &env);
```

Comment:
What is this buffer, who allocates it and who owns it? More API documentation needed.

Reply:
The buffer is needed for the network response, the caller allocate, owns and interprete this!

Reply:
How does the caller know how many bytes were written to that buffer? How does the caller know the content-type and the status code?